### ISSUE TITLE
[Bug]: Moving From 15.0.2 to 15.0.3, my application does not start, interceptor error

### ISSUE THREAD

**Issue by teyma** [damienbod/angular-auth-oidc-client#1786]
### Version

15.0.3

### Please provide the exception or error you saw

```true
in.ts:13 ERROR Error: Uncaught (in promise): TypeError: Cannot read properties of undefined (reading 'interceptor')
TypeError: Cannot read properties of undefined (reading 'interceptor')
    at handle (http.mjs:1443:21)
    at interceptRequest (angular-auth-oidc-client.mjs:4739:16)
    at AuthInterceptor.intercept (angular-auth-oidc-client.mjs:4713:16)
    at HttpInterceptorHandler.handle (http.mjs:1443:33)
    at HttpXsrfInterceptor.intercept (http.mjs:2046:25)
    at HttpInterceptorHandler.handle (http.mjs:1443:33)
    at HttpInterceptingHandler.handle (http.mjs:2094:27)
    at http.mjs:1270:70
    at doInnerSub (mergeInternals.js:19:19)
    at outerNext (mergeInternals.js:14:57)
    at resolvePromise (zone.js:1214:31)
    at resolvePromise (zone.js:1168:17)
    at zone.js:1281:17
    at _ZoneDelegate.invokeTask (zone.js:409:31)
    at Object.onInvokeTask (core.mjs:26194:33)
    at _ZoneDelegate.invokeTask (zone.js:408:60)
    at Zone.runTask (zone.js:178:47)
    at drainMicroTaskQueue (zone.js:588:35)
h
```

### Steps to reproduce the behavior

Moving from 15.0.2 to 15.0.3, I have an error in the console when I start my application.

My auth-config.module.ts

```true
const componentTest = () => navigator.userAgent.includes('component_test');

const authenticationFactory = (
  oidcSercurityService: OidcSecurityService,
  typescriptUtilisateurs: TypescriptUtilisateurs,
  http: HttpClient
) => {
  if (componentTest()) {
    return new TestAuthentication(http, typescriptUtilisateurs);
  }
  return new OidcAuthentication(oidcSercurityService, typescriptUtilisateurs);
};

const configurationFactory = (propertiesService: PropertiesService) => {
  const configurationOidcClient = propertiesService.loadProperties().pipe(
    map(properties => {
      return {
        secureRoutes: ['/api'],
        postLoginRoute: '/accueil',
        forbiddenRoute: '/forbidden',
        unauthorizedRoute: '/unauthorized',
        logLevel: LogLevel.Error,
        historyCleanupOff: false,
        authority: properties.oidc.issuerUri,
        redirectUrl: window.location.origin,
        postLogoutRedirectUri: window.location.origin + '/accueil',
        clientId: properties.oidc.clientId,
        scope: properties.oidc.scope.join(' '),
        responseType: 'code',
        silentRenew: true,
        useRefreshToken: true,
        ignoreNonceAfterRefresh: true,
      };
    })
  );

  return new StsConfigHttpLoader(configurationOidcClient);
};

@NgModule({
  declarations: [ForbiddenComponent, UnauthorizedComponent],
  imports: [
    AuthModule.forRoot({
      loader: {
        provide: StsConfigLoader,
        useFactory: configurationFactory,
        deps: [PropertiesService],
      },
    }),
  ],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: AuthInterceptor, multi: true },
    { provide: HTTP_INTERCEPTORS, useClass: AuthenticationInterceptor, multi: true },
    {
      provide: AUTHENTICATION,
      useFactory: authenticationFactory,
      deps: [OidcSecurityService, TypescriptUtilisateurs, HttpClient],
    },
  ],
  exports: [AuthModule],
})
export class AuthConfigModule {}
```

When I comment this line :

```
{ provide: HTTP_INTERCEPTORS, useClass: AuthInterceptor, multi: true },
```

the error disappears (I obviously get a new one from my backend, I no longer send the token).

### A clear and concise description of what you expected to happen.

```true
I wanted my application up :)
```

**Comment by teyma** [comment]
There it is : https://github.com/teyma/bug-angular-oidc-client

It seems it is coming from my other dependencies.

**Comment by FabianGosebrink** [comment]
So it is not connected with this library?

**Comment by teyma** [comment]
Yes it is, the error is in the angular-auth-oidc-client.mjs file

```
re.mjs:7635 ERROR Error: TypeError: Cannot read properties of undefined (reading 'interceptor')
    at angular-auth-oidc-client.mjs:3441:37
    at Observable.init [as _subscribe] (throwError.js:5:51)
    at Observable._trySubscribe (Observable.js:37:25)
    at Observable.js:31:30
    at errorContext (errorContext.js:19:9)
    at Observable.subscribe (Observable.js:22:21)
    at catchError.js:23:27
    at SafeSubscriber.<anonymous> (lift.js:10:28)
    at Observable.js:26:30
    at errorContext (errorContext.js:19:9)
h
```

But it might be because of some incompatibilities. That's my first guess.

**Comment by JustasJankauskasPersonal** [comment]
The same issue is on version 15.0.5.
When using AuthInterceptor in the providers as so:
```
{ provide: HTTP_INTERCEPTORS, useClass: AuthInterceptor, multi: true },
```

I get this error:
```
global-error.handler.ts:11 TypeError: Cannot read properties of undefined (reading 'interceptor')
    at handle (http.mjs:1442:21)
    at interceptRequest (angular-auth-oidc-client.mjs:4709:16)
    at AuthInterceptor.intercept (angular-auth-oidc-client.mjs:4683:16)
    at HttpInterceptorHandler.handle (http.mjs:1442:33)
    at HttpXsrfInterceptor.intercept (http.mjs:2048:25)
    at HttpInterceptorHandler.handle (http.mjs:1442:33)
    at HttpInterceptingHandler.handle (http.mjs:2094:27)
    at http.mjs:1269:70
    at doInnerSub (mergeInternals.js:19:19)
    at outerNext (mergeInternals.js:14:57)
```

When that interceptor is not used, application works.

**Comment by FabianGosebrink** [comment]
I did start your example but did not see the error. I went deeper, and you do not import the `HttpClientModule` but you are adding Interceptors. Provide both on the `app.module`.

**Comment by JustasJankauskasPersonal** [comment]
Actually the issue still exists even if HttpClientModule is imported. However, I am using Angular 14 at the moment while trying to use angular-auth-oidc-client version 15. Maybe that brings the issue? Can not so easily increment the version of angular so just asking here, maybe you have some quick answer based on internal knowledge about the library in general. :) Thanks.

**Comment by FabianGosebrink** [comment]
Can you do the sample so that I can see the error?

**Comment by stefanocke** [comment]
@FabianGosebrink , we have "similar" issues, and I found one change that is probably unrelated, but seem suspicious:

https://github.com/damienbod/angular-auth-oidc-client/commit/15566c20f3945ea6389e924aa59b893e07398107#diff-8273e285b2e825c1f8a550a8f35c7730f2b7f05a6b8d88fa51bbe7dd1835d49d

Here, `providedIn: root`  has been added to the AuthInterceptor.
Is this intentionally?
IMHO, an interceptor has always to be provided with the token `HTTP_INTERCEPTORS`. 
So, does it make sense to provide it in root?
